### DISCUSSION TITLE
monitor the K8S through the API Server

### DISCUSSION THREAD

**Discussion by dingzx3** [apache/skywalking#9739] [General]
To monitor the K8S, need to deploy skywalking in the K8S cluster? The source code does not show where the API Server address is set. Can I monitor the k8s status through the API Server address?

**Comment by wu-sheng** [comment]
That is a basic concept. you could learn from k8s. This is not SkyWalking's design.

**Comment by dingzx3** [comment]
Maybe I didn't make it clear. By modifying the ApiClient initialization method, we can monitor k8s cluster, k8s service, and k8s instance.
![image](https://user-images.githubusercontent.com/111035198/194701222-dc5f4afd-a87b-4c59-82dc-fe837a221158.png)

**Reply by wu-sheng** [reply]
This is not recommended by K8s mostly due to security concern.

API server should not be accessible from out of k8s.

**Comment by dingzx3** [comment]
![image](https://user-images.githubusercontent.com/111035198/194701265-b4149ef4-49e2-4438-8be3-afc2054a5d9f.png)
The main problem is that I cannot configure the apiserver address.
